update news

DyfanJones opened the pull request.

DyfanJones committed:
update news

DyfanJones commented:
Check `compute` function with latest version of `dbplyr`

```r
Last 13 lines of output:
    6.     └─dbplyr:::compute.tbl_sql(...)
    7.       ├─dbplyr::db_compute(...) at dbplyr/R/verb-compute.R:63:3
    8.       └─noctua::db_compute.AthenaConnection(...) at dbplyr/R/db-io.R:114:3
    9.         ├─dplyr::db_save_query(...) at noctua/R/dplyr_integration.R:104:5
   10.         └─dbplyr:::db_save_query.DBIConnection(...)
   11.           ├─dbplyr::sql_query_save(con, sql, name, temporary = temporary, ...) at dbplyr/R/db-sql.R:1126:3
   12.           └─rlang (local) `<fn>`() at dbplyr/R/db-sql.R:1126:3
   13.             └─rlang:::check_dots(env, error, action, call)
   14.               └─rlang:::action_dots(...)
   15.                 ├─base (local) try_dots(...)
   16.                 └─rlang (local) action(...)
  
  [ FAIL 4 | WARN 0 | SKIP 0 | PASS 337 ]
```

DyfanJones committed:
build athena ddl from sql query

DyfanJones committed:
update rdocumentation

pegeler commented:
Hi @DyfanJones, thanks for the great package!

The changes to `sql_query_save.AthenaConnection()` in this PR seem to fix an error I am experiencing with `dplyr::compute()`. When using the CRAN version of `noctua`, I get the error shown in the reprex below. However, the same pipeline with `noctua` 2.6.3 runs without errors.

Is there anything I can do to help get this fix on CRAN? I'm happy to do some digging if you point me in the right direction.

Thanks!

Paul

## Reprex

```r
con |>
  tbl("iris") |>
  group_by(species) |>
  summarise(n = n(), avg_sepal_length = mean(sepal_length)) |>
  compute(
    name = I("species_summary"),
    temporary = FALSE
  )
## Error in get(x, envir = ns, inherits = FALSE) : 
##   object 'as_from' not found
## Error in `sql_query_save()`:
## ! Arguments in `...` must be used.
## ✖ Problematic argument:
## • with = with
## ℹ Did you misspell an argument name?
## Backtrace:
##  1. dplyr::compute(...)
##  2. dbplyr:::compute.tbl_sql(...)
##  4. noctua::db_compute.AthenaConnection(...)
##  6. dbplyr:::db_save_query.DBIConnection(...)
##  7. dbplyr::sql_query_save(con, sql, name, temporary = temporary, ...)
```

## Session Info

```
R      4.4.1
DBI    1.2.3
dbplyr 2.5.0
dplyr  1.1.4
noctua 2.6.2
```

DyfanJones commented:
@pegeler sorry for the delay in this release i have been away on holiday of late. Plus the latest changes to paws has had me fairly busy.  I believe once the finally error has been resolve this can be release properly :)

pegeler committed:
fixing data.table::fread parsing when result file contains one column

DyfanJones committed:
Merge pull request #220 from pegeler/single-col-results

Fixing `data.table::fread` parsing when result file contains one column

pegeler commented on R/dplyr_integration.R:
```diff
@@ -126,17 +126,20 @@ athena_query_save <- function(con, sql, name, with, ...) {
+  if (isTRUE(temporary)) {
+    stop("AWS Athena doesn't support temporary tables.")
+  }
+  name <- DBI::dbQuoteIdentifier(con, name)
```
Hi @DyfanJones,

This `DBI::dbQuoteIdentifier()` call is causing me some issues when the caller has already quoted the name, for example `dplyr::compute()` passes in a `table_path` object that has already been quoted via [`db_save_query.DBIConnection()`](https://github.com/tidyverse/dbplyr/blob/572693098d0098018b6bb2d1e88b650d2e1e1580/R/db-sql.R#L1125). So quoting again causes too many quotes and also makes it impossible to save in a schema other than the default schema.

Could the function either provide a `no_quote` parameter or check to see if the `name` already inherits from a class where we can be reasonably sure that quoting should not be performed? (Such as `table_path`, `AsIs`, dplyr's `sql`) I know this could be a sensitive issue because of the security concerns around SQL injection. But it would be helpful to have an escape hatch since right now I can't think of a way to get this to work with `dplyr::compute()` when saving across schemas.

Thanks again for the excellent package and I look forward to your perspective on this issue.

Paul

DyfanJones committed:
fix dplyr integration

DyfanJones committed:
support engineversion

DyfanJones committed:
fix conflict

Merge branch 'noctua_2.6.3' of github.com:DyfanJones/noctua into noctua_2.6.3

# Conflicts:
#	R/File_Parser.R

DyfanJones committed:
set up rhub

DyfanJones committed:
revert dynamic sep

DyfanJones committed:
prevent result dropping to the lowest possible dimension

DyfanJones committed:
Explicitly link DBI classes
